libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -166,10 +166,6 @@ func (d *addDockerMetadata) Run(event *beat.Event) (*beat.Event, error) {
 	container := d.watcher.Container(cid)
 	if container != nil {
 		meta := common.MapStr{}
-		metaIface, ok := event.Fields["docker"]
```

Comment:
Remove this one as I felt it's not needed anymore when using `Put` with the DeepUpdate. Please double check if that is correct.

Reply:
while that's true, I think that will create more unneeded allocations? this is the kind of code that runs lots of times

Reply:
Not sure I can fully follow here. You are mostly worried about the `DeepUpdate` part?

Reply:
Going to merge for now. @exekias Let's revisit this next year to make sure we have the correct optimisations in place.